### PR TITLE
ContainerSelectorTest: Refactoring and bug fixing

### PR DESCRIPTION
**Pull request by Garfield96**
# Description

Remove code duplication in ContainerSelectorTest and correct calculation of halo box. Before, not all faces of the halo box were tested.

# How Has This Been Tested?
- [x] Jenkins

### CODE REVIEW

**Comment by FG-TUM** [comment]
@Steffen without looking at the code, does this overlap with your rework of this test?

**Comment by SteffenSeckler** [comment]
seems fine to me, I didn't really change existing tests, so there shouldn't be too much overlap

File: tests/testAutopas/tests/selectors/ContainerSelectorTest.cpp [hunk 1]
```diff
@@ -32,11 +32,29 @@ TEST_F(ContainerSelectorTest, testSelectAndGetCurrentContainer) {
   }
 }
 
+void getStatus(const std::array<double, 3> &bBoxMin, const std::array<double, 3> &bBoxMax, const double cutoff,
```
Comment (FG-TUM) [inline tests/testAutopas/tests/selectors/ContainerSelectorTest.cpp]: doc/comment what this does would help for future maintainability

File: tests/testAutopas/tests/selectors/ContainerSelectorTest.cpp [hunk 2]
```diff
@@ -32,11 +32,29 @@ TEST_F(ContainerSelectorTest, testSelectAndGetCurrentContainer) {
...
+               autopas::ContainerSelector<Particle, FPCell> &containerSelector, std::vector<Particle> &ListInner,
+               std::vector<Particle> &ListHalo, std::vector<Particle> &ListHaloVerletOnly) {
+  for (auto iter = containerSelector.getCurrentContainer()->begin(autopas::IteratorBehavior::ownedOnly); iter.isValid();
+       ++iter) {
+    ListInner.push_back(*iter);
+  }
+  for (auto iter = containerSelector.getCurrentContainer()->begin(autopas::IteratorBehavior::haloOnly); iter.isValid();
+       ++iter) {
+    if (autopas::utils::inBox(iter->getR(), autopas::ArrayMath::sub(bBoxMin, std::array<double, 3>{cutoff}),
+                              autopas::ArrayMath::add(bBoxMax, std::array<double, 3>{cutoff}))) {
```
Comment (FG-TUM) [inline tests/testAutopas/tests/selectors/ContainerSelectorTest.cpp]: for readability, you could move the array calculations to separate variables like in lines 76,77.

**Pull request merged**